src/lib/db/feature-toggle-store.ts

```diff
@@ -377,6 +396,40 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
 
         return toggle;
     }
+
+    async markPotentiallyStaleFeatures(
+        currentTime?: string,
+    ): Promise<string[]> {
+        const query = this.db(TABLE)
+            .update('potentially_stale', true)
+            .whereRaw(
+                `? > (features.created_at + ((
+                    SELECT feature_types.lifetime_days
+                    FROM feature_types
+                    WHERE feature_types.id = features.type
+                ) * INTERVAL '1 day'))`,
+                [currentTime || this.db.fn.now()],
+            )
+            .andWhere(function () {
+                this.where('potentially_stale', null).orWhere(
+                    'potentially_stale',
+                    false,
+                );
+            })
+            .andWhereNot('stale', true);
+
+        const updatedFeatures = await query.returning(FEATURE_COLUMNS);
+        return updatedFeatures.map(({ name }) => name);
+    }
+
+    async isPotentiallyStale(featureName: string): Promise<boolean> {
+        const result = await this.db(TABLE)
+            .first([...FEATURE_COLUMNS, 'potentially_stale'])
```

Comment:
I think we dont need to return more columns than `potentially_stale`

Reply:
Agreed! 5e5a3d1c9a253740682eef4bad537e4bd0ab2949